xarray/core/formatting.py

```diff
@@ -300,11 +300,18 @@ def _summarize_coord_multiindex(coord, col_width, marker):
 
 
 def _summarize_coord_levels(coord, col_width, marker="-"):
+    if col_width < len(coord):
+        n_values = col_width // 4
+        indices = list(range(0, n_values)) + list(range(-n_values, 0))
+        subset = coord[indices]
+    else:
+        subset = coord
```

Comment:
this could probably use some optimization: how big does the `MultiIndex` have to be so indexing+`get_level_variable` is faster than just `get_level_variable`?

Reply:
Yeah, though it's so fast already relative to how often repr is called...

Could also defer to pandas, which seem to do this (though a different orientation)

More than fine to leave as a TODO imo

Reply:
I assumed that for everything below 100 elements `get_level_variable` is faster than indexing first, which also means that I don't have to worry about changing the resulting `repr`.

> Could also defer to pandas

how would I do that?

Reply:
> how would I do that?

I had meant — they have a repr for multiindex which is fast — so could we use theirs somehow, despite the different orientation. On reflection — our code is simple, I agree with your impulse.